src/uucore/src/lib/features/quoting_style.rs

```diff
@@ -205,144 +253,182 @@ impl Iterator for EscapedChar {
     }
 }
 
-fn shell_without_escape(name: &str, quotes: Quotes, show_control_chars: bool) -> (String, bool) {
-    let mut must_quote = false;
-    let mut escaped_str = String::with_capacity(name.len());
+/// Check whether `bytes` starts with any byte in `pattern`.
+fn bytes_start_with(bytes: &[u8], pattern: &[u8]) -> bool {
```

Comment:
What about inlining this function ? I don't know if we have guidelines regarding inlining, but this seems to be a correct candidate.

Reply:
As in `#[inline]`? I'd rather let the optimizer figure that out, unless we have benchmarks demonstrating a need otherwise.

Reply:
Yes, I was thinking of `#[inline]` precisely